File: 0026-remove-duplicates-from-sorted-array/0026-remove-duplicates-from-sorted-array.py
class Solution(object):
    def removeDuplicates(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
# i           j
# 0, 0, 0, 0, 1, 1, 1, 2, 2, 3, 3, 4 
        i=0

        for j in range (1, len(nums)): 
            if nums[i]!=nums[j]: 
                nums[i+1]= nums[j]
                i+=1
        return i+1

File: 0026-remove-duplicates-from-sorted-array/0026-remove-duplicates-from-sorted-array.java
class Solution {
    public int removeDuplicates(int[] nums) {
        int j=0; 
        int i=1;
        
        while(i<nums.length){
            if(nums[i-1]!=nums[i]){
                nums[j+1]=nums[i]; 
                i++; 
                j++; 
            }
            else{
                i++; 
            }
        }
        return j+1;
    }
}